features/step_definitions/logging.rb

```diff
@@ -1056,3 +1056,120 @@
+
+Given /^cluster-admin create the #{QUOTED} pattern for the #{WORD} user$/ do | pattern_name, who |
+    user(word_to_num(who))
+    user_name=user.name
+    user_token=user.cached_tokens.first
+    step %Q/I switch to cluster admin pseudo user/
+    step %Q/I use the "openshift-logging" project/
+    
+    rest_cmd="curl -s --connect-timeout 10  -XPOST \"http://localhost:5601/api/saved_objects/index-pattern/#{pattern_name}\" -H \"kbn-xsrf: true\" -H \"x-forwarded-user: #{user_name}\" -H \"securitytenant: __user__\" -H \"Authorization: Bearer #{user_token}\" -H \"Content-Type: application/json\" -d \'{\"attributes\": {\"title\": \"#{pattern_name}\", \"timeFieldName\": \"@timestamp\"}}\'"
+
+    step %Q/a pod becomes ready with labels:/, table(%{
+        | component= kibana|
+    })
+    @result = pod.exec("bash", "-c", rest_cmd, as: admin, container: 'kibana')
+    raise "Failed to create index pattern via rest api" unless @result[:success]
+end
+
+Given /^I can display the(?: "(.+)")? pod logs of the(?: "(.+)")? project under the(?: "(.+)")? pattern in kibana$/ do | pod_name, project_name, pattern_name |
+    # I switch to cluster admin pseudo user
+    pod_name ||= "*"
+    project_name ||= "*"
+    pattern_name ||= "app"
+    #lucene_query_string = "kubernetes.namespace_name: #{project_name} and kubernetes.pod_name: #{pod_name} and @timestamp:[now-15m TO now]"
+    lucene_query_string = "kubernetes.namespace_name: #{project_name} and kubernetes.pod_name: #{pod_name}"
+    step %Q/I perform the :select_index_pattern_in_kibana web action with:/,table(%{
+             | index_pattern_name | #{pattern_name} |
+    })
+    raise("Can not select to pattern #{ pattern_name}!") unless @result[:success]
+
+    step %Q/I perform the :search_doc_in_kibana web action with:/,table(%{
+             | search_string | #{lucene_query_string} |
+    })
+    raise("Can not refresh the search #{lucene_query_string}!") unless @result[:success]
+
+    retries = 3
+    while true
+        step %Q/I run the :check_log_count web action/
+        if @result[:success]
+           break
+        elsif retries > 0
+           log("Can not find documents using clause #{lucene_query_string} under pattern  #{ pattern_name}! retry.....")
+        else
+           raise("Can not find documents using clause #{lucene_query_string} under pattern  #{ pattern_name}! after 3 retry,  abort")
+        end
+        step %Q/I run the :click_refresh_button_in_kibana action/
+        retries -= 1
+    end
+end
+
+Given /^I have clusterlogging with(?: (\d+))? persistent storage ES$/ do |es_num|
+    # I switch to cluster admin pseudo user
+    ensure_admin_tagged
+    es_num ||= 3
+    redundancy_policy="SingleRedundancy"
+    deploy_cluster_logging=true
+
+    if cluster_logging('instance').exists?
+       deploy_cluster_logging=false
+       unless cluster_logging('instance').logstore_storage.has_key?("storageClassName")
+           deploy_cluster_logging=true
+       end
+    end
```

Comment:
Per my understanding, this block can be simplified to:
```
       if cluster_logging('instance').logstore_storage.has_key?("storageClassName")
           deploy_cluster_logging=false
       end
```

Reply:
No. I got error below using step above


    Given I have clusterlogging with persistent storage ES                                                     # features/step_definitions/logging.rb:1169
      [06:36:23] INFO> Shell Commands: oc get clusterloggings instance --output=yaml --kubeconfig=/home/anli/workdir/laptop-anli-e/ocp4_admin.kubeconfig --namespace=logging-upg-prep-share
      
      STDERR:
      Error from server (NotFound): clusterloggings.logging.openshift.io "instance" not found
      
      [06:36:25] INFO> Exit Status: 1
      clusterloggings 'instance' not found (BushSlicer::ResourceNotFoundError)
      /home/anli/cucushift-runtime/verification-tests/lib/openshift/resource.rb:63:in `get_checked'
      /home/anli/cucushift-runtime/verification-tests/lib/openshift/resource.rb:127:in `get_cached_prop'
      /home/anli/cucushift-runtime/verification-tests/lib/openshift/resource.rb:134:in `raw_resource'
      /home/anli/cucushift-runtime/verification-tests/lib/openshift/cluster_logging.rb:132:in `log_store_spec'
      /home/anli/cucushift-runtime/verification-tests/lib/openshift/cluster_logging.rb:157:in `logstore_storage'
      /home/anli/cucushift-runtime/verification-tests/features/step_definitions/logging.rb:1183:in `/^I have clusterlogging with(?: (\d+))? persistent storage ES$/'
      features/upgrade/logging/upgrade.feature:16:in `I have clusterlogging with persistent storage ES'

Reply:
I saw `--namespace=logging-upg-prep-share`, be careful with the namespace